admin/worker/billing_reporter.go

```diff
@@ -0,0 +1,115 @@
+	if err != nil {
+		return fmt.Errorf("failed to get last usage reporting time: %w", err)
+	}
+
+	// start reporting from the last reported time
+	var startTime time.Time
+	if t.IsZero() {
+		startTime = time.Now().UTC().Truncate(granularity).Add(-granularity)
+	} else {
+		startTime = t.UTC()
+	}
+	// report until end of the last hour
+	endTime := time.Now().UTC().Truncate(granularity)
+
+	if !startTime.Before(endTime) {
+		w.logger.Debug("skipping usage reporting: no new usage data available", zap.Time("start_time", startTime), zap.Time("end_time", endTime))
+		return nil
+	}
+
+	// Get metrics client
+	client, ok, err := w.admin.OpenMetricsProject(ctx)
+	if err != nil {
+		w.logger.Error("failed to report usage: unable to get metrics client", zap.Error(err))
+		return err
+	}
+	if !ok {
+		w.logger.Debug("skipping usage reporting: no metrics project configured")
+		return nil
+	}
+
+	u, err := client.GetUsageMetrics(ctx, startTime, endTime, sqlGrainIdentifier)
+	if err != nil {
+		return fmt.Errorf("failed to get usage metrics: %w", err)
+	}
+
+	if len(u) == 0 {
+		w.logger.Warn("skipping usage reporting: no usage data available", zap.Time("start_time", startTime), zap.Time("end_time", endTime))
+		return nil
+	}
+
+	reportedOrgs := make(map[string]struct{})
+	var usage []*billing.Usage
+	for _, m := range u {
+		reportedOrgs[m.OrgID] = struct{}{}
+		usage = append(usage, &billing.Usage{
+			CustomerID:     m.OrgID,
+			MetricName:     m.MetricName,
+			Amount:         m.Amount,
+			ReportingGrain: w.admin.Biller.GetReportingGranularity(),
+			StartTime:      m.StartTime,
+			EndTime:        m.EndTime,
+			Metadata:       map[string]interface{}{"org_id": m.OrgID, "project_id": m.ProjectID},
+		})
+	}
+
+	err = w.admin.Biller.ReportUsage(ctx, usage)
+	if err != nil {
+		return fmt.Errorf("failed to report usage: %w", err)
+	}
+
+	// update last reporting time to endTime as it is the max processed event time
+	err = w.admin.DB.UpdateBillingUsageReportedOn(ctx, endTime)
+	if err != nil {
+		return fmt.Errorf("failed to update last usage reporting time: %w", err)
+	}
+
+	// get orgs which have billing customer id
+	orgs, err := w.admin.DB.FindBillingOrganizationIDs(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to report usage: unable to fetch orgs: %w", err)
+	}
+
+	// get orgs which have billing customer id and not reported in this run
+	for _, org := range orgs {
+		if _, ok := reportedOrgs[org]; !ok {
+			// count the projects which are not hibernated and created before the given time
+			count, err := w.admin.DB.CountBillingProjectsForOrganization(ctx, org, endTime)
+			if err != nil {
+				w.logger.Warn("failed to validate active projects for org", zap.String("org", org), zap.Error(err))
+				continue
+			}
+			if count > 0 {
+				w.logger.Warn("skipping usage reporting for org: no usage data available", zap.String("org", org), zap.Time("start_time", startTime), zap.Time("end_time", endTime))
+			}
+		}
+	}
```

Comment:
(Not needed for this PR) To avoid loading all orgs into memory at once, and to speed up the job, we could consider pulling this logic into a separate job. The new job could be a kind of "billing usage health check job" that maybe even just calls Orb directly and checks if usage data is available for within the last few hours for each org with billing projects.